src/merge.js

```diff
@@ -22,23 +22,19 @@ function mergeOne(dst, src) {
   // See if 'dst' is allowed to be mutated. If not - it's overridden with a new plain object.
   const returnValue = isObject(dst) ? dst : {};
 
-  const keys = Object.keys(src);
-  for (let i = 0; i < keys.length; i++) {
-    const key = keys[i];
-
+  return Object.keys(src).reduce((result, key) => {
```

Comment:
Sorry. That's the right link: https://reaktor.com/blog/javascript-performance-fundamentals-make-bluebird-fast/